packages/jsdoc-plugins/lib/statics-augmentor/statics-augmentor.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * @license Copyright (c) 2017, CKSource - Frederico Knabben. All rights reserved.
+ * Licensed under the terms of the MIT License (see LICENSE.md).
+ */
+
+'use strict';
+
+/**
+ * Manually adds doclets for inherited static members.
+ * Jsdoc doesn't support inheritance of static members which is why this plugin is needed.
+ */
+class StaticsAugmentor {
+	constructor( doclets ) {
+		this._data = doclets;
+
+		this._longnames = doclets.index.longname;
+
+		this._documented = doclets.index.documented;
+	}
+
+	/**
+	 * Main method responsible for inheriting static members.
+	 */
+	augmentStatics() {
+		const childClasses = this._getDescendants();
+
+		for ( let child of childClasses ) {
+			const parents = child.augments;
+
+			for ( let parent of parents ) {
+				const staticMembers = this._getStaticMembers( parent );
+
+				for ( let member of staticMembers ) {
+					this._handleStaticMember( member, child );
+				}
+			}
+		}
+	}
+
+	/**
+	 * Finds doclets of child classes which need to have inherited static members added.
+	 * These have to be of kind `class` or `interface` and have a non empty `augments` array.
+	 * @returns {Array.<Doclet>}
+	 * @private
+	 */
+	_getDescendants() {
+		return this._data.filter( doclet => Array.isArray( doclet.augments ) &&
+			doclet.augments.length > 0 &&
+			( doclet.kind === 'class' || doclet.kind === 'interface' )
+		);
+	}
+
+	/**
+	 * Gets all static members of a doclet.
+	 * @param longname
+	 * @returns {Array.<Doclet>}
+	 * @private
+	 */
+	_getStaticMembers( longname ) {
+		return this._data.filter( doclet => doclet.memberof === longname && doclet.scope === 'static' );
+	}
+
+	/**
+	 * @param {Doclet }member
+	 * @param {Doclet} childClass
+	 * @private
+	 */
+	_handleStaticMember( member, childClass ) {
+		if ( member.undocumented || member.ignore ) {
+			return;
+		}
+		// deep clone member
+		const clone = JSON.parse( JSON.stringify( member ) );
+
+		this._addNewDoclet( clone, member, childClass );
+	}
+
+	/**
+	 * Modifies and adds new doclet to array of doclets. Makes existing doclets ignored if needed.
+	 * @param {Doclet} doclet New doclet which will be added to doclets array.
+	 * @param {Doclet} original Doclet from which new doclet is created.
+	 * @param {Doclet} childClass Doclet of a child class.
+	 * @private
+	 */
+	_addNewDoclet( doclet, original, childClass ) {
+		let parts;
+
+		if ( !doclet.inherited ) {
+			doclet.inherits = doclet.longname;
+		}
+		doclet.inherited = true;
+
+		doclet.memberof = childClass.longname;
+
+		// static members are separated by a dot
```

Comment:
All comments should start with upper case letter and should be ending with a dot.